### ISSUE TITLE
`no-useless-undefined`: should work when a function has a return type specified

### ISSUE THREAD

**Issue by cdavie-artium** [sindresorhus/eslint-plugin-unicorn#2747]
When enabling the `no-useless-undefined` rule in Typescript, I expect both of the following to get flagged:

```ts
function getsFlaggedCorrectly() {
  return undefined; // 👍 flagged
}

function shouldBeFlaggedButIsNot(): undefined {
  return undefined; // ☹️ this SHOULD be flagged, but is NOT
}
```
